firewood/src/storage/linear/proposed.rs

```diff
@@ -53,20 +51,11 @@ impl<P: ReadLinearStore, M: Send + Sync + Debug> ReadLinearStore for Proposed<P,
     }
 }
 
-/// Marker that the Proposal is mutable
-#[derive(Debug)]
-pub(crate) struct Mutable;
-
-/// Marker that the Proposal is immutable
-#[derive(Debug)]
-pub(crate) struct Immutable;
-
-impl<P: ReadLinearStore> WriteLinearStore for Proposed<P, Mutable> {
+impl WriteLinearStore for Proposed {
```

Comment:
Needs to be bound by mutability. I don't want *all* `Proposed` types to have the `write` method.

Reply:
Same response as in https://github.com/ava-labs/firewood/pull/633#discussion_r1574805130 -- doesn't the use (or lack of use) of `mut` already give us a way of controlling when a `Proposed` is `write`able?

Reply:
https://github.com/ava-labs/firewood/pull/633#discussion_r1575131164